src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,74 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
-
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
-
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+import { useEffect, useRef, useState } from 'react';
+import { Todo } from '../../types/Todo';
+import { getDisplayedTodos } from '../../utis';
+type Props = {
+  todos: Todo[];
+  updateTodos: (todos: Todo[]) => void;
+};
+
+export const TodoFilter: React.FC<Props> = ({ todos, updateTodos }) => {
+  const [query, setQuery] = useState('');
+  const [filter, setFilter] = useState('all');
+  const timerID = useRef(0);
+
+  useEffect(() => {
+    let displayedTodos = getDisplayedTodos(todos, filter, '');
+
+    updateTodos(displayedTodos);
+
+    if (query) {
+      timerID.current = window.setTimeout(() => {
+        displayedTodos = getDisplayedTodos(todos, filter, query);
+
+        updateTodos(displayedTodos);
+      }, 1000);
+    }
+  }, [todos, filter, query]);
```

Comment:
looks too complicated 

just move all states from this component into App and filter the array there without logic with useEffect and updateTodos